service/history/historyEngine.go

```diff
@@ -1608,7 +1609,7 @@ func (e *historyEngineImpl) RequestCancelWorkflowExecution(
 		RunId:      request.WorkflowExecution.RunId,
 	}
 
-	return e.updateWorkflowExecution(ctx, domainID, execution, true,
+	return e.updateWorkflowExecutionWithContextReload(ctx, domainID, execution, true,
```

Comment:
can we unify updateWorkflowExecutionWithContextReload and updateWorkflowExecutionWithActionAndContextReload?

Reply:
I can just remove `updateWorkflowExecutionWithContextReload`, but it would cause caller to call something like 
```
updateWorkflowExecutionWithActionAndContextReload(ctx, domainID, execution,
		getUpdateWorkflowActionFunc(createDecisionTask, action))
```

Since we already use things like `updateWorkflowExecution` and `updateWorkflowExecutionWithAction`, I think it's better to have this retry context version `updateWorkflowExecutionWithContextReload` and `updateWorkflowExecutionWithActionAndContextReload`

Reply:
we already have too many wrapped version of update workflow execution, can we simplify / unify that?